lock-impl/src/main/java/com/palantir/lock/impl/LegacyTimelockService.java

```diff
@@ -103,23 +106,28 @@ public long getImmutableTimestamp() {
     }
 
     @Override
-    public LockRefreshTokenV2Adapter lock(LockRequestV2 request) {
+    public Optional<LockTokenV2> lock(LockRequestV2 request) {
         LockRequest legacyRequest = toLegacyLockRequest(request);
 
         LockRefreshToken legacyToken = lockAnonymous(legacyRequest);
-        return new LockRefreshTokenV2Adapter(legacyToken);
+        return Optional.ofNullable(legacyToken).map(LockRefreshTokenV2Adapter::new);
     }
 
     @Override
-    public void waitForLocks(WaitForLocksRequest request) {
+    public boolean waitForLocks(WaitForLocksRequest request) {
         LockRequest legacyRequest = toLegacyWaitForLocksRequest(request.getLockDescriptors());
 
         lockAnonymous(legacyRequest);
+        return true;
```

Comment:
I'm not sure what a good solution to this in terms of readability would be though, a comment perhaps?